lib/board.rb

```diff
@@ -21,20 +22,26 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
-    if coordinates.length != ship.length
+    if overlap?(coordinates) == true
+      false
+    elsif coordinates.length != ship.length
       false
     elsif consecutive_coordinates?(coordinates) == false
       false
-    elsif diagonal_coordinates?(coordinates)
+    elsif diagonal_coordinates?(coordinates) == true
```

Comment:
Needed something to compare the method to or it would always return false regardless of the outcome.

Reply:
Nice! I know we had it working but it did feel incomplete!